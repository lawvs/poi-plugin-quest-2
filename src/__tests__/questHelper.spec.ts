import {
  calcQuestMap,
  getCompletedQuest,
  getLockedQuest,
  getPostQuestIds,
  getPreQuestIds,
  getQuestCodeByGameId,
} from '../questHelper'

describe('questHelper', () => {
  test('should getQuestCodeByGameId correct', () => {
    expect(getQuestCodeByGameId(0)).toEqual(null)
    expect(getQuestCodeByGameId(101)).toEqual('A1')
  })

  test('should getPreQuestIds correct', () => {
    expect(getPreQuestIds(101)).toEqual([])
    expect(getPreQuestIds(102)).toEqual([101])
    expect(getPreQuestIds(236)).toEqual([235, 273])
  })

  test('should getPostQuestIds correct', () => {
    expect(getPostQuestIds(101)).toEqual([102])
    expect(getPostQuestIds(105)).toEqual([106, 108, 174, 254, 401, 612])
    expect(getPostQuestIds(140)).toEqual([])
  })

  test('should 101 no completed quest', () => {
    expect(getCompletedQuest([101])).toEqual({})
  })

  test('should getCompletedQuest quest match snapshot', () => {
    expect(calcQuestMap([817], getPreQuestIds)).toMatchSnapshot()
  })

  test('should 236 getCompletedQuest correct', () => {
    expect(calcQuestMap([236], getPreQuestIds)).toMatchSnapshot()
  })

  test('should 101 locked quests match snapshot', () => {
    expect(getLockedQuest([101])).toMatchSnapshot()
  })

  test('should 196 getLockedQuest correct', () => {
    expect(getLockedQuest([196])).toMatchSnapshot()
  })
})
