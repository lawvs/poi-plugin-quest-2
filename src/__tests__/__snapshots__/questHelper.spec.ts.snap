// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`questHelper should 101 locked quests match snapshot 1`] = `
{
  "1001": true,
  "1003": true,
  "1005": true,
  "1006": true,
  "1007": true,
  "1008": true,
  "1013": true,
  "102": true,
  "103": true,
  "104": true,
  "105": true,
  "106": true,
  "107": true,
  "108": true,
  "109": true,
  "110": true,
  "1101": true,
  "1102": true,
  "1103": true,
  "1104": true,
  "1105": true,
  "1106": true,
  "1108": true,
  "1109": true,
  "111": true,
  "1111": true,
  "1113": true,
  "1116": true,
  "1117": true,
  "1118": true,
  "112": true,
  "1120": true,
  "1127": true,
  "1128": true,
  "1129": true,
  "113": true,
  "1130": true,
  "1131": true,
  "1132": true,
  "1133": true,
  "1134": true,
  "1137": true,
  "1138": true,
  "1139": true,
  "114": true,
  "115": true,
  "116": true,
  "117": true,
  "118": true,
  "119": true,
  "120": true,
  "121": true,
  "122": true,
  "123": true,
  "124": true,
  "125": true,
  "126": true,
  "127": true,
  "128": true,
  "129": true,
  "130": true,
  "131": true,
  "132": true,
  "133": true,
  "134": true,
  "135": true,
  "136": true,
  "137": true,
  "138": true,
  "139": true,
  "140": true,
  "141": true,
  "142": true,
  "143": true,
  "144": true,
  "145": true,
  "146": true,
  "147": true,
  "148": true,
  "149": true,
  "150": true,
  "151": true,
  "152": true,
  "153": true,
  "154": true,
  "155": true,
  "156": true,
  "157": true,
  "158": true,
  "161": true,
  "162": true,
  "163": true,
  "164": true,
  "165": true,
  "166": true,
  "167": true,
  "168": true,
  "169": true,
  "170": true,
  "171": true,
  "172": true,
  "173": true,
  "174": true,
  "175": true,
  "176": true,
  "177": true,
  "178": true,
  "179": true,
  "180": true,
  "181": true,
  "182": true,
  "183": true,
  "184": true,
  "185": true,
  "186": true,
  "187": true,
  "188": true,
  "189": true,
  "190": true,
  "191": true,
  "192": true,
  "193": true,
  "194": true,
  "195": true,
  "196": true,
  "197": true,
  "198": true,
  "215": true,
  "217": true,
  "219": true,
  "222": true,
  "223": true,
  "224": true,
  "225": true,
  "227": true,
  "231": true,
  "232": true,
  "233": true,
  "236": true,
  "237": true,
  "239": true,
  "240": true,
  "244": true,
  "245": true,
  "246": true,
  "247": true,
  "248": true,
  "249": true,
  "250": true,
  "251": true,
  "252": true,
  "253": true,
  "254": true,
  "258": true,
  "259": true,
  "260": true,
  "261": true,
  "262": true,
  "263": true,
  "264": true,
  "265": true,
  "266": true,
  "267": true,
  "268": true,
  "269": true,
  "270": true,
  "271": true,
  "273": true,
  "274": true,
  "275": true,
  "276": true,
  "277": true,
  "278": true,
  "279": true,
  "281": true,
  "282": true,
  "285": true,
  "287": true,
  "288": true,
  "289": true,
  "290": true,
  "291": true,
  "292": true,
  "293": true,
  "294": true,
  "295": true,
  "296": true,
  "297": true,
  "298": true,
  "306": true,
  "307": true,
  "308": true,
  "309": true,
  "312": true,
  "316": true,
  "318": true,
  "319": true,
  "323": true,
  "324": true,
  "325": true,
  "327": true,
  "328": true,
  "330": true,
  "331": true,
  "337": true,
  "339": true,
  "342": true,
  "344": true,
  "346": true,
  "347": true,
  "350": true,
  "351": true,
  "355": true,
  "356": true,
  "357": true,
  "358": true,
  "360": true,
  "362": true,
  "364": true,
  "365": true,
  "366": true,
  "368": true,
  "371": true,
  "372": true,
  "401": true,
  "402": true,
  "403": true,
  "404": true,
  "405": true,
  "406": true,
  "408": true,
  "409": true,
  "410": true,
  "411": true,
  "412": true,
  "413": true,
  "414": true,
  "415": true,
  "418": true,
  "419": true,
  "420": true,
  "422": true,
  "423": true,
  "424": true,
  "425": true,
  "427": true,
  "428": true,
  "429": true,
  "431": true,
  "433": true,
  "434": true,
  "437": true,
  "439": true,
  "440": true,
  "442": true,
  "443": true,
  "444": true,
  "445": true,
  "610": true,
  "611": true,
  "612": true,
  "614": true,
  "615": true,
  "616": true,
  "618": true,
  "619": true,
  "620": true,
  "621": true,
  "622": true,
  "623": true,
  "624": true,
  "625": true,
  "626": true,
  "627": true,
  "628": true,
  "629": true,
  "630": true,
  "631": true,
  "632": true,
  "633": true,
  "634": true,
  "635": true,
  "636": true,
  "638": true,
  "639": true,
  "641": true,
  "642": true,
  "643": true,
  "644": true,
  "645": true,
  "646": true,
  "647": true,
  "648": true,
  "649": true,
  "650": true,
  "651": true,
  "652": true,
  "654": true,
  "655": true,
  "656": true,
  "658": true,
  "660": true,
  "661": true,
  "662": true,
  "663": true,
  "664": true,
  "665": true,
  "666": true,
  "667": true,
  "669": true,
  "670": true,
  "671": true,
  "672": true,
  "676": true,
  "681": true,
  "683": true,
  "684": true,
  "685": true,
  "686": true,
  "687": true,
  "688": true,
  "689": true,
  "690": true,
  "695": true,
  "696": true,
  "698": true,
  "704": true,
  "718": true,
  "719": true,
  "805": true,
  "806": true,
  "807": true,
  "808": true,
  "809": true,
  "811": true,
  "812": true,
  "813": true,
  "814": true,
  "815": true,
  "817": true,
  "818": true,
  "819": true,
  "820": true,
  "821": true,
  "822": true,
  "823": true,
  "824": true,
  "825": true,
  "826": true,
  "827": true,
  "828": true,
  "829": true,
  "833": true,
  "835": true,
  "836": true,
  "837": true,
  "842": true,
  "844": true,
  "845": true,
  "846": true,
  "847": true,
  "848": true,
  "850": true,
  "851": true,
  "852": true,
  "854": true,
  "856": true,
  "858": true,
  "859": true,
  "860": true,
  "861": true,
  "862": true,
  "863": true,
  "864": true,
  "865": true,
  "869": true,
  "870": true,
  "871": true,
  "872": true,
  "873": true,
  "874": true,
  "875": true,
  "877": true,
  "879": true,
  "881": true,
  "884": true,
  "885": true,
  "886": true,
  "887": true,
  "888": true,
  "889": true,
  "890": true,
  "891": true,
  "892": true,
  "896": true,
  "897": true,
  "898": true,
  "901": true,
  "902": true,
  "903": true,
  "904": true,
  "905": true,
  "911": true,
  "912": true,
  "913": true,
  "914": true,
  "915": true,
  "916": true,
  "917": true,
  "918": true,
  "924": true,
  "925": true,
  "926": true,
  "927": true,
  "928": true,
  "929": true,
  "930": true,
  "931": true,
  "933": true,
  "934": true,
  "935": true,
  "936": true,
  "937": true,
  "938": true,
  "939": true,
  "940": true,
  "943": true,
  "947": true,
  "948": true,
  "949": true,
  "958": true,
  "961": true,
  "966": true,
  "967": true,
  "968": true,
  "969": true,
  "970": true,
  "973": true,
  "974": true,
  "975": true,
  "976": true,
  "977": true,
  "978": true,
  "980": true,
  "981": true,
  "982": true,
  "983": true,
  "989": true,
  "993": true,
  "995": true,
  "996": true,
  "998": true,
}
`;

exports[`questHelper should 196 getLockedQuest correct 1`] = `
{
  "197": true,
  "346": true,
  "892": true,
  "930": true,
}
`;

exports[`questHelper should 236 getCompletedQuest correct 1`] = `
{
  "101": true,
  "102": true,
  "103": true,
  "105": true,
  "106": true,
  "107": true,
  "113": true,
  "119": true,
  "121": true,
  "124": true,
  "125": true,
  "131": true,
  "133": true,
  "136": true,
  "152": true,
  "202": true,
  "203": true,
  "204": true,
  "205": true,
  "206": true,
  "227": true,
  "235": true,
  "273": true,
  "307": true,
}
`;

exports[`questHelper should getCompletedQuest quest match snapshot 1`] = `
{
  "101": true,
  "102": true,
  "103": true,
  "105": true,
  "174": true,
}
`;
